airflow/providers/microsoft/azure/triggers/data_factory.py

```diff
@@ -89,3 +89,119 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:
                 await asyncio.sleep(self.poke_interval)
         except Exception as e:
             yield TriggerEvent({"status": "error", "message": str(e)})
+
+
+class AzureDataFactoryTrigger(BaseTrigger):
+    """
+    AzureDataFactoryTrigger is triggered when Azure data factory pipeline job succeeded or failed.
+    When wait_for_termination is set to False it triggered immediately with success status
+
+    :param run_id: Run id of a Azure data pipeline run job.
+    :param azure_data_factory_conn_id: The connection identifier for connecting to Azure Data Factory.
+    :param end_time: Time in seconds when triggers will timeout.
+    :param resource_group_name: The resource group name.
+    :param factory_name: The data factory name.
+    :param wait_for_termination: Flag to wait on a pipeline run's termination.
+    :param check_interval: Time in seconds to check on a pipeline run's status.
+    """
+
+    QUEUED = "Queued"
+    IN_PROGRESS = "InProgress"
+    SUCCEEDED = "Succeeded"
+    FAILED = "Failed"
+    CANCELING = "Canceling"
+    CANCELLED = "Cancelled"
+
+    INTERMEDIATE_STATES: list[str] = [QUEUED, IN_PROGRESS, CANCELING]
+    FAILURE_STATES: list[str] = [FAILED, CANCELLED]
+    SUCCESS_STATES: list[str] = [SUCCEEDED]
+    TERMINAL_STATUSES: list[str] = [CANCELLED, FAILED, SUCCEEDED]
```

Comment:
[`AzureDataFactoryPipelineRunStatus`](https://github.com/apache/airflow/blob/f6472c83eb5c50b567d51e3a8869df5442f6953c/airflow/providers/microsoft/azure/hooks/data_factory.py#L111) already exists so let's use that. No problem augmenting the status groups too with  `INTERMEDIATE_STATUSES` and `FAILURE_STATUSES`. I don't think `SUCCESS_STATES` is needed though since the logic could be `pipeline_status == AzureDataFactoryPipelineRunStatus.SUCCESS`. `TERMINAL_STATUSES` already exists in AzureDataFactoryPipelineRunStatus.